changelog/@unreleased/pr-4656.v2.yml

```diff
@@ -0,0 +1,6 @@
+type: improvement
+improvement:
+  description: |
+    The retained memory in delete executor work queue is reduced. Previously, each runnable in deleteExecutor had SnapshotTransaction reference, which can be potentially huge.
```

Comment:
minor nits/technicalities: the reference to a SnapshotTransaction can't be huge, it's just the word size. Maybe instead something like `which could mean a lot of memory was unnecessarily retained`?